python/semanage/seobject.py

```diff
@@ -1070,13 +1070,19 @@ def __genkey(self, port, proto):
         if port == "":
             raise ValueError(_("Port is required"))
 
-        ports = port.split("-")
+        if isinstance(port, str):
+            ports = port.split('-', 1)
+        else:
+            ports = (port,)
+            pass
+        
         if len(ports) == 1:
             high = low = int(ports[0])
         else:
             low = int(ports[0])
             high = int(ports[1])
-
+            pass
+        
```

Comment:
Yes, why are you adding pass statements?

Reply:
It makes for proper code formatting when you cut-n-paste or move blocks of code. It compiles out, and I add such things without thinking.  I have been writing in python for so long that I forget why I do half the things I do.

I will remove it and resub.